### PR TITLE
Improve readme + keep text when linter/completion mode is changing

### PR DESCRIPTION
**Pull request by Nexucis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,28 @@
-CodeMirror-mode-promql
-======================
+CodeMirror-promql
+=================
 [![CircleCI](https://circleci.com/gh/prometheus-community/codemirror-promql.svg?style=shield)](https://circleci.com/gh/prometheus-community/codemirror-promql) [![GitHub license](https://img.shields.io/badge/license-MIT-blue.svg)](./LICENSE)
 [![NPM version](https://img.shields.io/npm/v/codemirror-promql.svg)](https://www.npmjs.org/package/codemirror-promql)
 
 ## Overview
-This project provides a mode for [CodeMirror Next](https://codemirror.net/6) that handles syntax highlighting and autocompletion for the PromQL ([Prometheus Query Language](https://prometheus.io/docs/introduction/overview/)).
+This project provides a mode for [CodeMirror Next](https://codemirror.net/6) that handles syntax highlighting, linter and autocompletion for the PromQL ([Prometheus Query Language](https://prometheus.io/docs/introduction/overview/)).
```
Comment (juliusv) [inline README.md:7]:
```suggestion
This project provides a mode for [CodeMirror Next](https://codemirror.net/6) that handles syntax highlighting, linting, and autocompletion for the PromQL ([Prometheus Query Language](https://prometheus.io/docs/introduction/overview/)).
```

File: README.md [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
 
 ### Status
 This mode is **not** production ready. If you want to use it, you may encounter some issues.
+API and usage of this mode can be deeply changed between two versions.
 
 See the [Development](#development) section for more information on features and usage.
 
 ### Installation
-This mode is available as an npm package:
+This mode is available as a npm package:
```
Comment (juliusv) [inline README.md:19]:
```suggestion
This mode is available as an npm package:
```

File: README.md [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
 
-| Version             | command to use                        |
-| ------------------- | ------------------------------------- |
-| >= 0.3.0            | `npm install codemirror-promql`       |
-| >= 0.2.2 < 0.3.0    | `npm install codemirror-mode-promql`  |
+```bash
+npm install codemirror-promql
+```
+
+**Note:** You will have to install on your own side [CodeMirror Next](https://codemirror.net/6)
```
Comment (juliusv) [inline README.md:25]:
```suggestion
**Note:** You will have to manually install [CodeMirror Next](https://codemirror.net/6) as a peer dependency.
```

File: README.md [hunk 4]
```diff
@@ -67,10 +72,20 @@ rest_api_port: 8000
...
+ **Note:** To avoid CORS issues when testing LSP-based completion, the development web server running on port 8080 proxies LSP requests to the language server backend running on port 8000. Alternatively, if you decide to remove this proxying and want the browser to be able to speak to your LSP backend directly, start your browser with web security disabled.
  * Example for Google Chrome on Windows:
     * `<Win>` + `r`  and copy and paste `chrome.exe --user-data-dir="C:/Chrome dev session" --disable-web-security`
 
+### Linter
+
+The linter feature has 2 different modes, each requiring a different setup:
+
+ * **lsp**: This mode requires the same setup as described before
+ * **offline**: this mode doesn't require anything
```
Comment (juliusv) [inline README.md:84]:
```suggestion
 * **offline**: This mode doesn't require anything
```

File: README.md [hunk 5]
```diff
@@ -67,10 +72,20 @@ rest_api_port: 8000
...
+ 
+ **Note:** Changing the linter mode at the runtime required to reload entirely CodeMirror.
```
Comment (juliusv) [inline README.md:86]:
```suggestion
 **Note:** Changing the linter mode during runtime requires reloading CodeMirror entirely. 
```

File: README.md (hunk 3 above)
Comment (dvkndn) [reply README.md:25]: @juliusv I think the "as a peer dependency" may not be exactly what you mean. CodeMirror Next is the peer dependency of this library, so the user must install CMN themselves, but as a normal dependency, not as a peer dependency (of the user's app)?

Comment (juliusv) [reply README.md:25]: Oh yeah, good point. So maybe just:

```suggestion
**Note:** You will have to manually install [CodeMirror Next](https://codemirror.net/6), as it is a peer dependency to this package.
```

**Comment by Nexucis** [comment]
thanks for the correction from both of view @juliusv and @dvkndn .

**Pull request merged**